libdnf/conf/config_main.cpp

```diff
@@ -186,6 +187,9 @@ class ConfigMain::Impl {
     OptionBool debug_solver{false};
     OptionStringList installonlypkgs{INSTALLONLYPKGS};
     OptionStringList group_package_types{GROUP_PACKAGE_TYPES};
+    OptionStringSet metadata_types{flags_to_strings(
+        libdnf::repo::LoadFlags::PRIMARY | libdnf::repo::LoadFlags::MODULES | libdnf::repo::LoadFlags::COMPS |
```

Comment:
I believe that `libdnf::repo::LoadFlags::PRIMARY | libdnf::repo::LoadFlags::MODULES` should be hardcoded types - not in option.

Reply:
I used to have these values "hardcoded" when converting the string list in the `LoadFlags::flags_from_strings()`. I can move it there again, but I think that we cannot really prevent user bypassing these metadata types if they really want to. We'll probably have to override getting the values from the `Option`.

Reply:
Oh but when I thinking about it now, we don't have any conditional loading for `PRIMARY` and `MODULES` now. It is always loaded, the `MODULES` could be bypassed only with compile flag. So I think we could just omit these here if it will be more transparent?

Reply:
his is a good question. We can keep those value there but we have to document that removal of those values has no effect on workflow. But I am not sure what would be best. The requirements is to document the behavior in documentation.

Reply:
I think the best place would be the documentation for configuration options which is not present yet. But I've added a comment about this into our internal JIRA task.